web/static/js/reducers/questionnaire.js

```diff
@@ -150,28 +166,28 @@ type ActionChangeStepSmsPrompt = {
   newPrompt: string
 };
 
-const changeStepSmsPrompt = (state, action: ActionChangeStepSmsPrompt) => {
+const changeStepSmsPrompt = (state, action: ActionChangeStepSmsPrompt, quiz: Questionnaire): Step[] => {
   return changeStep(state, action.stepId, step => ({
     ...step,
     prompt: {
       ...step.prompt,
-      'en': {
-        ...step.prompt['en'],
+      [quiz.defaultLanguage]: {
+        ...step.prompt[quiz.defaultLanguage],
         sms: action.newPrompt
       }
     }
   }))
 }
 
-const changeStepIvrPrompt = (state, action) => {
+const changeStepIvrPrompt = (state, action, quiz: Questionnaire) => {
   return changeStep(state, action.stepId, step => ({
     ...step,
     prompt: {
       ...step.prompt,
-      'en': {
-        ...step.prompt['en'],
+      [quiz.defaultLanguage]: {
```

Comment:
Again, this is just a patch. The right solution would be to remove the `quiz` param and add a `lang` prop to the action that can then be used to address the right content.